twoliter/src/project/lock/mod.rs

```diff
@@ -200,6 +204,43 @@ impl Lock {
         }
     }
 
+    fn validate_version_source_consistency(&self) -> Result<()> {
```

Comment:
Could you also add docstring comments to these functions, telling us what they validate? It would be great to read the function with an understanding of what it is guarding against.